documentation/docs/tutorials/setup.md

```diff
@@ -528,3 +528,142 @@ The Grafana dashboard should be accessible at `http://<your-ip>:3000`.
 Default login credentials are:
 * `username`: admin
 * `password`: admin
+## Installing Goshimmer by Building From Source
+### Software Requirements
+Upgrade your systems' packages by running the following command:
+```bash
+apt update && apt dist-upgrade -y
+```  
+
+#### Installing RocksDB Compression Libraries
+
+GoShimmer uses RocksDB as its underlying database engine. You can install the dependencies required by RocksDB by running the following command:
+
+```bash
+sudo apt-get install libgflags-dev libsnappy-dev zlib1g-dev libbz2-dev liblz4-dev libzstd-dev
+```  
+
+
+####  GCC and G++
+
+GCC and G++ are required for the compilation to work properly.  You can install them by running the following command:
+
+```bash
+sudo apt install gcc g++
+```
+
+### Clone the Repository
+
+Once you have installed the [software requirements](#software-requirements), you should clone the [GoShimmer repository](https://github.com/iotaledger/goshimmer/) into the `/opt` directory. You can do so by running the following commands: 
+
+```bash
+cd /opt
+git clone https://github.com/iotaledger/goshimmer.git
+```
+
+###  Golang-go 1.16
+
+Your system will need to have [Go 1.16](https://go.dev/doc/go1.16) installed for the build script to work. You can install it by running the following command:
+
+```bash
+apt install golang-go
+````
+
+You can make sure you are using the right go version by running the following command:
+
+```bash
+go version
+```
+
+If apt did not install Go version 1.16+, you can use the following commands to download and install it from source:
+
+
+```bash
+cd /home
+wget https://dl.google.com/go/go1.16.4.linux-amd64.tar.gz
+sudo tar -xvf go1.16.4.linux-amd64.tar.gz
+sudo mv go /usr/local
+export GOROOT=/usr/local/go
+cd /opt/goshimmer
+mkdir gopath
+export GOPATH=/opt/goshimmer/gopath
+export PATH=$GOPATH/bin:$GOROOT/bin:$PATH
+```
+
+You run `go version` again to check if you have successfully installed `golang-go 1.16`.
+This installation is only valid for this session. If you enter a new session, you will have to enter these commands again:
+
+```bash
+export GOROOT=/usr/local/go
+export GOPATH=/opt/goshimmer/gopath
+export PATH=$GOPATH/bin:$GOROOT/bin:$PATH
+```
+
```

Comment:
Is there a way to make the installation permanent? If so, it may be good to add it here.

Reply:
Do not know how to yet. Is on my to-do list.

Reply:
Generally you put these environment variables in the shell's config file so that it can load when you start a session. But this depends on which shell a user is using, e.g., `.bashrc` or `.zshrc`. Here's some info on how to do it: https://linuxhint.com/export-a-path-in-bashrc/

However, I think this is way too detailed here.

Reply:
Ok, removed this part.